mle/model.py

```diff
@@ -156,6 +156,91 @@ def stream(self, chat_history, **kwargs):
                 yield delta.content
 
 
+class ClaudeModel(Model):
+    def __init__(self, api_key, model, temperature=0.7):
+        """
+        Initialize the Claude model.
+        Args:
+            api_key (str): The Anthropic API key.
+            model (str): The model with version.
+            temperature (float): The temperature value.
+        """
+        super().__init__()
+
+        dependency = "anthropic"
+        spec = importlib.util.find_spec(dependency)
+        if spec is not None:
+            self.anthropic = importlib.import_module(dependency).Anthropic
+        else:
+            raise ImportError(
+                "It seems you didn't install anthropic. In order to enable the OpenAI client related features, "
+                "please make sure openai Python package has been installed. "
+                "More information, please refer to: https://docs.anthropic.com/en/api/client-sdks"
+            )
+
+        self.model = model if model else 'claude-3-5-sonnet-20240620'
+        self.model_type = MODEL_CLAUDE
+        self.temperature = temperature
+        self.client = self.anthropic(api_key=api_key)
+
+    def query(self, chat_history, **kwargs):
+        """
+        Query the LLM model.
+
+        Args:
+            chat_history: The context (chat history).
+        """
+        # claude has not system role in chat_history
+        # https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/system-prompts
+        system_prompt = ""
+        for idx, msg in enumerate(chat_history):
+            if msg["role"] == "system":
+                system_prompt = msg["content"]
+                chat_history.pop(idx)
+
+        # claude does not support mannual `response_format`, so we append it into system prompt
+        if "response_format" in kwargs.keys():
+            system_prompt += f"\n You must output in {kwargs['response_format']['type']} format"
+
+        completion = self.client.messages.create(
+            max_tokens=4096,
+            model=self.model,
+            system=system_prompt,
+            messages=chat_history,
+            temperature=self.temperature,
+            stream=False
+        )
+
+        return completion.content[0].text
+
+    def stream(self, chat_history, **kwargs):
+        """
+        Stream the output from the LLM model.
+        Args:
+            chat_history: The context (chat history).
+        """
+        # claude has not system role in chat_history
+        # https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/system-prompts
+        system_prompt = ""
+        for idx, msg in enumerate(chat_history):
+            if msg["role"] == "system":
+                system_prompt = msg["content"]
+                chat_history.pop(idx)
+
+        # claude does not support mannual `response_format`, so we append it into system prompt
```

Comment:
Hi, @huangyz0918. I found that Claude cannot generate content strictly in JSON format, even if we add the format requirements into the system prompt. Any thoughts or experiences with this?

```
claude repsonse:
Certainly! I'll analyze the user requirement and dataset, then provide a recommendation based on the latest research 
and best practices. Let me start by gathering some information.

{
    "task": "Image Classification",
    "model_or_algorithm": "ResNet8",
    "frameworks": ["PyTorch", "TensorFlow"],
    "reference": [
        "https://arxiv.org/abs/1512.03385",
        "https://arxiv.org/abs/1706.02677"
    ],
    "evaluation_metric": ["Accuracy", "F1-score"],
    "training_method": "Supervised Learning with Data Augmentation",
    "serving_method": "Model can be served using TensorFlow Serving or TorchServe",
    "device": "GPU (recommended for faster training)",
    "data_summary": "The MNIST dataset is a widely used benchmark dataset for handwritten digit recognition. It 
contains 60,000 training images and 10,000 test images of handwritten digits (0-9), each 28x28 pixels in grayscale.",
    "suggestion": "Based on the user requirement to train ResNet8 on the MNIST dataset, we suggest the following..."
}

Traceback (most recent call last):
  File "/root/miniconda3/envs/mle/bin/mle", line 33, in <module>
    sys.exit(load_entry_point('mle-agent', 'console_scripts', 'mle')())
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/root/miniconda3/envs/mle/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/root/workspace/MLE-agent/mle/cli.py", line 73, in start
    return baseline(os.getcwd(), model)
  File "/root/workspace/MLE-agent/mle/workflow/baseline.py", line 69, in baseline
    advisor_report = advisor.interact("[green]User Requirement:[/green] " + ml_requirement + "\n" + ask_data(dataset))
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 131, in interact
    self.report = self.suggest(requirement)
  File "/root/workspace/MLE-agent/mle/agents/advisor.py", line 121, in suggest
    suggestions = json.loads(text)
  File "/root/miniconda3/envs/mle/lib/python3.10/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/root/miniconda3/envs/mle/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/root/miniconda3/envs/mle/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

Reply:
Please refer to this https://docs.anthropic.com/en/docs/test-and-evaluate/strengthen-guardrails/increase-consistency

Reply:
Thanks a lot! I just prefilled the assistant output to constrain its output JSON format. The workflow went well in my tests.
![image](https://github.com/user-attachments/assets/92fceb6e-273e-4b45-9a26-1aa22fabc78c)
refer to: https://docs.anthropic.com/en/docs/test-and-evaluate/strengthen-guardrails/increase-consistency